### PR TITLE
fix(app-router): flush server-inserted HTML during SSR streaming

### PR DESCRIPTION
**Pull request by NathanDrake2406**
## What this changes
App Router SSR now asks for `useServerInsertedHTML` output on each HTML streaming flush instead of only once before returning the stream. The first insertion still places the navigation/bootstrap/font/server HTML before `</head>`, while later insertions emit newly collected server-inserted HTML before the next streamed chunk.

## Why
CSS-in-JS registries such as styled-components and Emotion register a callback during shell render, then collect additional styles as Suspense boundaries continue rendering. Vinext previously flushed and cleared those callbacks before the stream was consumed, so styles collected after the shell never reached streamed HTML.

Next.js keeps server-inserted HTML callbacks registered and asks for insertion HTML from its stream transform repeatedly:

- [`createServerInsertedHTML` stores callbacks without clearing them](https://github.com/vercel/next.js/blob/ae61573e062e900050b8e6b24626e450accc4570/packages/next/src/server/app-render/server-inserted-html.tsx#L9-L30)
- [`makeGetServerInsertedHTML` renders the registered callbacks each time it is called](https://github.com/vercel/next.js/blob/ae61573e062e900050b8e6b24626e450accc4570/packages/next/src/server/app-render/make-get-server-inserted-html.tsx#L38-L109)
- [`createHeadInsertionTransformStream` invokes insertion during stream transforms and final flush](https://github.com/vercel/next.js/blob/ae61573e062e900050b8e6b24626e450accc4570/packages/next/src/server/stream-utils/node-web-streams-helper.ts#L447-L520)
- [Next.js e2e coverage asserts Suspense CSS-in-JS styles stream before the refresh script](https://github.com/vercel/next.js/blob/ae61573e062e900050b8e6b24626e450accc4570/test/e2e/app-dir/use-server-inserted-html/use-server-inserted-html.test.ts#L50-L68)
- [The referenced fixture renders a styled-components node inside Suspense](https://github.com/vercel/next.js/blob/ae61573e062e900050b8e6b24626e450accc4570/test/e2e/app-dir/use-server-inserted-html/app/css-in-js/suspense/page.js#L31-L58)

## Approach
- Add `renderServerInsertedHTML()` for streaming SSR so callbacks can be invoked without unregistering them.
- Keep callback cleanup deferred until the returned SSR stream is consumed or cancelled.
- Let `createTickBufferedTransform()` accept an insertion function and call it after the initial head injection on each buffered flush.
- Preserve the existing one-shot string insertion path for current non-streaming callers.

## Validation
- `vp check`
- `vp test run tests/server-inserted-html.test.ts tests/server-inserted-html-context.test.ts tests/rsc-streaming.test.ts tests/app-router.test.ts tests/features.test.ts tests/app-page-stream.test.ts`

## Risks / follow-ups
The change is scoped to App Router SSR streaming and the existing tick-buffered transform. The test coverage is unit and integration-level; it ports the Next.js streaming order assertion without adding a full styled-components dependency fixture to vinext.

### CODE REVIEW

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and test gaps.

**Comment by NathanDrake2406** [comment]
/bigbonk review this PR. Focus on blocking correctness issues and whether the tests cover the real behavior and meaningful edge cases.

File: packages/vinext/src/server/app-ssr-stream.ts [hunk 1]
```diff
@@ -96,22 +98,34 @@ export function fixPreloadAs(html: string): string {
...
 
   const flushBuffered = (controller: TransformStreamDefaultController<Uint8Array>): void => {
+    if (buffered.length === 0) return;
+
+    if (injected && insertsPerFlush) {
+      const insertion = readInsertion();
+      if (insertion) {
+        controller.enqueue(encoder.encode(insertion));
+      }
+    }
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/server/app-ssr-stream.ts:125]: Added the ordering comment. The insertion-before-buffered-HTML invariant is now explicit in app-ssr-stream.ts.

File: packages/vinext/src/server/app-ssr-stream.ts [hunk 2]
```diff
@@ -153,8 +167,12 @@ export function createTickBufferedTransform(
 
       flushBuffered(controller);
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/server/app-ssr-stream.ts:173]: Verified this edge case and fixed it. The transform now emits a final per-flush insertion in flush() when the initial head insertion already happened, even if the last timeout drained the buffered HTML before stream close. Added a regression test for that exact drained-buffer path.
